packages/drivers/routerlicious-driver/src/historian.ts

```diff
@@ -0,0 +1,200 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { fromUtf8ToBase64 } from "@fluidframework/common-utils";
+import * as git from "@fluidframework/gitresources";
+import {
+	IWholeFlatSummary,
+	IWholeSummaryPayload,
+	IWriteSummaryResponse,
+} from "@fluidframework/server-services-client";
+import { QueryStringType, RestWrapper } from "./restWrapperBase";
+import { IR11sResponse } from "./restWrapper";
+import { IHistorian } from "./storageContracts";
+
+function endsWith(value: string, endings: string[]): boolean {
+	for (const ending of endings) {
+		if (value.endsWith(ending)) {
+			return true;
+		}
+	}
+
+	return false;
+}
+
+export interface ICredentials {
+	user: string;
+	password: string;
+}
+
+export const getAuthorizationTokenFromCredentials = (credentials: ICredentials): string =>
+	`Basic ${fromUtf8ToBase64(`${credentials.user}:${credentials.password}`)}`;
+
+/**
+ * Implementation of the IHistorian interface that calls out to a REST interface
+ */
+export class Historian implements IHistorian {
+	private readonly defaultQueryString: QueryStringType = {};
+	private readonly cacheBust: boolean;
+
+	constructor(
+		public endpoint: string,
+		private readonly historianApi: boolean,
+		disableCache: boolean,
+		private readonly restWrapper: RestWrapper,
+	) {
+		if (disableCache && this.historianApi) {
+			this.defaultQueryString.disableCache = disableCache;
+			this.cacheBust = false;
+		} else {
+			this.cacheBust = disableCache;
+		}
+	}
+
+	public async getHeader(sha: string): Promise<any> {
+		return this.historianApi
+			? this.restWrapper.get(`/headers/${encodeURIComponent(sha)}`, this.getQueryString())
+			: this.getHeaderDirect(sha);
+	}
+
+	public async getFullTree(sha: string): Promise<any> {
+		return this.restWrapper.get(`/tree/${encodeURIComponent(sha)}`, this.getQueryString());
+	}
+
+	public async getBlob(sha: string): Promise<git.IBlob> {
+		return this.restWrapper.get<git.IBlob>(
+			`/git/blobs/${encodeURIComponent(sha)}`,
+			this.getQueryString(),
+		);
+	}
+
+	public async createBlob(blob: git.ICreateBlobParams): Promise<git.ICreateBlobResponse> {
+		return this.restWrapper.post<git.ICreateBlobResponse>(
+			`/git/blobs`,
+			blob,
+			this.getQueryString(),
+		);
+	}
+
+	public async getContent(path: string, ref: string): Promise<any> {
+		return this.restWrapper.get(`/contents/${path}`, this.getQueryString({ ref }));
+	}
+
+	public async getCommits(sha: string, count: number): Promise<git.ICommitDetails[]> {
```

Comment:
same question as in previous file - do we actually use all the public methods in driver code? I think the answer is no and we should do some cleanup